typescript-config.js

```diff
@@ -0,0 +1,184 @@
+    },
+    linterOptions: {
+      reportUnusedDisableDirectives: true,
+    },
+  },
+  {
+    ignores: ['public/**/*.*', 'vendor/**/*.*', '*.js'],
+  },
+  {
+    plugins: {
+      '@stylistic/ts': stylisticTs,
+      '@typescript-eslint': tsPlugin,
+      'eslintProgressRule': {
+        rules: { showProgress: eslintProgressRule }
+      }
+    },
+    rules: {
+      'eslintProgressRule/showProgress': 1,
+      quotes: ['error', 'single'],
+      indent: ['error', 2],
+      semi: ['error', 'never'],
+      'comma-dangle': ['error', 'always-multiline'],
+      'eol-last': ['error', 'always'],
+      'object-curly-spacing': ['error', 'always'],
+      'tailwindcss/classnames-order': 'off',
+      'tailwindcss/no-custom-classname': 'off',
+      'n/no-missing-import': 'off',
+      'n/no-extraneous-import': 'off',
+      'no-undef': 'off',
+      'no-void': ['error', { allowAsStatement: true }],
+      'no-unused-vars': 'off',
+      'no-return-await': 'off',
+      'n/no-unsupported-features/node-builtins': 'off',
+    },
+    linterOptions: {
+      reportUnusedDisableDirectives: true,
+    },
+  },
+  {
+    files: ['**/*.vue'],
+    languageOptions: {
+      parser: vueParser,
+      parserOptions: {
+        parser: tsParser,
+        ecmaVersion: 'latest',
+        sourceType: 'module',
+        globals: globals.node,
+      },
+    },
+    rules: {
+      'vue/multi-word-component-names': 'off',
+      'vue/first-attribute-linebreak': 'off',
+      'vue/max-attributes-per-line': 'off',
+      'vue/singleline-html-element-content-newline': 'off',
+      'vue/padding-line-between-blocks': ['error', 'always'],
+    }
+  },
+  {
+    files: ['**/*.ts'],
+    languageOptions: {
+      parser: tsParser,
+      parserOptions: {
+        project: true,
+        ecmaVersion: 'latest',
+      },
+    },
+    rules: {
+      '@typescript-eslint/await-thenable': 'error',
+      '@typescript-eslint/consistent-type-exports': ['error', {
+        fixMixedExportsWithInlineTypeSpecifier: true,
+      }],
+      '@typescript-eslint/no-base-to-string': 'error',
+      '@typescript-eslint/no-for-in-array': 'error',
+      '@typescript-eslint/no-implied-eval': 'error',
+      '@typescript-eslint/no-misused-promises': 'error',
+      '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
+      '@typescript-eslint/no-unnecessary-type-assertion': 'error',
+      '@typescript-eslint/prefer-includes': 'error',
+      '@typescript-eslint/prefer-readonly': 'error',
+      '@typescript-eslint/prefer-reduce-type-parameter': 'error',
+      '@typescript-eslint/promise-function-async': 'error',
+      '@typescript-eslint/require-array-sort-compare': ['error', { ignoreStringArrays: true }],
+      '@typescript-eslint/restrict-plus-operands': ['error', { skipCompoundAssignments: false }],
+      '@typescript-eslint/restrict-template-expressions': ['error', { allowNumber: true }],
+      '@typescript-eslint/return-await': ['error', 'always'],
+    }
```

Comment:
I think this can be a source of weird issues and misunderstanding

Reply:
fixed